sormas-backend/src/main/java/de/symeda/sormas/backend/event/EventGroupFacadeEjb.java

```diff
@@ -411,6 +418,45 @@ public void linkEventsToGroups(List<String> eventUuids, List<String> eventGroupU
 		}
 	}
 
+	public List<EventReferenceDto> getEligibleEventUuidsToBeLinkedToGroup(List<String> eventUuids, List<String> ineligibleEventUuids) {
+		List<String> eligibleEventUuids = ineligibleEventUuids.size() > 0
+			? eventUuids.stream().filter(uuid -> !ineligibleEventUuids.contains(uuid)).collect(Collectors.toCollection(ArrayList::new))
+			: eventUuids;
+
+		List<EventReferenceDto> eligibleEventReferences = new ArrayList<>();
+		eligibleEventUuids.forEach(uuid -> eligibleEventReferences.add(new EventReferenceDto(uuid)));
+
+		return eligibleEventReferences;
```

Comment:
you can do map instead of forEach and add items to the list

Reply:
I also found a bug in the logic I wrote for the events which were already linked to the event groups (the already linked events were not displayed that the linking was not possible) and I've fixed the bug.